crates/warpgate/src/loader.rs

```diff
@@ -150,6 +156,102 @@ impl PluginLoader {
         .await
     }
 
+    /// Load a plugin from an OCI registry.
+    #[instrument(skip(self))]
+    pub async fn load_plugin_from_oci<I: AsRef<Id> + Debug>(
+        &self,
+        id: I,
+        locator: &warpgate_api::OciLocator,
+        registries: Vec<OciRegistry>,
+    ) -> Result<PathBuf, WarpgateLoaderError> {
+        let id = id.as_ref();
+        let client = Client::new(ClientConfig::default());
```

Comment:
Add a `get_oci_client` similar to `get_client`. Maybe rename `get_client` to `get_http_client`.